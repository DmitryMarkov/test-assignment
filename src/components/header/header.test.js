import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import Header, { headerSelector } from './'

const initialState = {
  offers: {
    metaData: {
      cursor: {
        totalCount: 2,
      },
      listTitle: 'hotels',
      searchRegion: {
        name: 'Test',
      },
    },
  },
}

const mockStore = configureStore([])

const withRedux = ({ children }) => (
  <Provider store={mockStore(initialState)}>{children}</Provider>
)

test('Header should render properly', async () => {
  const { asFragment, findByText } = render(<Header />, { wrapper: withRedux })
  const headingTitle = await findByText(/Test/)

  expect(headingTitle).toBeInTheDocument()
  expect(asFragment()).toMatchSnapshot()
})

test('headerSelector return proper values', () => {
  const returnedObject = {
    listTitle: 'hotels',
    name: 'Test',
    totalCount: 2,
  }

  expect(headerSelector(initialState, initialState.offers.metaData)).toEqual(
    returnedObject
  )
})
