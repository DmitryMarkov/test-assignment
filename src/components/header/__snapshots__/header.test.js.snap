// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `
<DocumentFragment>
  <h1
    class="css-15icvz4"
    data-testid="heading-title"
  >
    Test: 2 hotels
  </h1>
</DocumentFragment>
`;
